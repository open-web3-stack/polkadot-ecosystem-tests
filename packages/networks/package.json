{
	"name": "@e2e-test/networks",
	"type": "module",
	"dependencies": {
		"@acala-network/chopsticks-testing": "^0.13.1",
		"@e2e-test/shared": "workspace:*"
	},
	"main": "./dist/cjs/index.js",
	"module": "./dist/esm/index.js",
	"files": [
		"*",
		"!tsconfig.json",
		"!**/*.test.ts"
	],
	"exports": {
		".": {
			"require": "./dist/cjs/index.js",
			"import": "./dist/esm/index.js",
			"default": "./dist/esm/index.js"
		},
		"./chains": {
			"require": "./dist/cjs/chains/index.js",
			"import": "./dist/esm/chains/index.js",
			"default": "./dist/esm/chains/index.js"
		},
		"./*": {
			"require": "./dist/cjs/*.js",
			"import": "./dist/esm/*.js",
			"default": "./dist/esm/*.js"
		}
	}
}
