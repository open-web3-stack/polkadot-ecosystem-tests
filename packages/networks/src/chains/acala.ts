import { defaultAccount } from '@e2e-test/shared'

import { defineChain } from '../defineChain.js'

const custom = {
  acala: {
    relayToken: 'DOT',
    relayLiquidToken: 'LDOT',
    stableToken: 'AUSD',

    paraAccount: '13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy',
    dot: { Token: 'DOT' },
    ldot: { Token: 'LDOT' },
    dai: { Erc20: '0x54a37a01cd75b616d63e0ab665bffdb0143c52ae' },
    wbtc: { ForeignAsset: 5 },
    ausd: { Token: 'AUSD' },
    aca: { Token: 'ACA' },
    lcdot: { LiquidCrowdloan: 13 },
  },
  karura: {
    relayToken: 'KSM',
    relayLiquidToken: 'LKSM',
    stableToken: 'KUSD',

    paraAccount: '13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy',
    ksm: { Token: 'KSM' },
    lksm: { Token: 'LKSM' },
    usdt: { ForeignAsset: 7 },
    rmrk: { ForeignAsset: 0 },
    dai: { Erc20: '0x4bb6afb5fa2b07a5d1c499e1c3ddb5a15e709a71' },
    ausd: { Token: 'KUSD' },
    kar: { Token: 'KAR' },
  },
}

const getInitStorages = (config: typeof custom.acala | typeof custom.karura) => ({
  System: {
    account: [[[defaultAccount.alice.address], { providers: 4, data: { free: 10 * 1e12 } }]],
  },
  Tokens: {
    accounts: [
      [[defaultAccount.alice.address, { Token: config.relayToken }], { free: 10 * 1e12 }],
      [[defaultAccount.alice.address, { Token: config.relayLiquidToken }], { free: 100 * 1e12 }],
      [[defaultAccount.alice.address, { Token: config.stableToken }], { free: 1000 * 1e12 }],
    ],
  },
  Sudo: {
    key: defaultAccount.alice.address,
  },
  EvmAccounts: {
    accounts: [[['0x82a258cb20e2adb4788153cd5eb5839615ece9a0'], defaultAccount.alice.address]],
    evmAddresses: [[[defaultAccount.alice.address], '0x82a258cb20e2adb4788153cd5eb5839615ece9a0']],
  },
  Homa: {
    // avoid impact test outcome
    $removePrefix: ['redeemRequests', 'unbondings', 'toBondPool'],
    // so that bump era won't trigger unbond
    relayChainCurrentEra: 100,
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers', 'supportedVersion'],
  },
})

export const acala = defineChain({
  name: 'acala',
  endpoint: 'wss://acala-rpc.aca-api.network',
  paraId: 2000,
  custom: custom.acala,
  initStorages: getInitStorages(custom.acala),
})

export const karura = defineChain({
  name: 'karura',
  endpoint: 'wss://karura-rpc.aca-api.network',
  paraId: 2000,
  custom: custom.karura,
  initStorages: getInitStorages(custom.karura),
})
