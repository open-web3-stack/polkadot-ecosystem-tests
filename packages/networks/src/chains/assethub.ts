import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubPolkadot: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
  assetHubKusama: {
    ksm: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubPolkadot | typeof custom.assetHubKusama) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }]],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubPolkadot = defineChain({
  name: 'assetHubPolkadot',
  endpoint: 'wss://rpc-asset-hub-polkadot.luckyfriday.io',
  paraId: 1000,
  custom: custom.assetHubPolkadot,
  initStorages: getInitStorages(custom.assetHubPolkadot),
})

export const assetHubKusama = defineChain({
  name: 'assetHubKusama',
  endpoint: 'wss://rpc-asset-hub-kusama.luckyfriday.io',
  paraId: 1000,
  custom: custom.assetHubKusama,
  initStorages: getInitStorages(custom.assetHubKusama),
})
