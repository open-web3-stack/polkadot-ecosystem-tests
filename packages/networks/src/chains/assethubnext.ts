import { defaultAccounts, defaultAccountsSr25519 } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubNextWestend: {
    wnd: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubNextWestend) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 1000e10 } }],
    ],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubNextWestend = defineChain({
  name: 'assetHubNextWestend',
  endpoint: 'wss://westend-asset-hub-next-rpc.parity-chains-scw.parity.io',
  paraId: 1100,
  custom: custom.assetHubNextWestend,
  initStorages: getInitStorages(custom.assetHubNextWestend),
})
