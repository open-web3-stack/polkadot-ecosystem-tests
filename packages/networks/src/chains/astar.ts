import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  astar: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551617',

    paraAccount: '13YMK2eZzuFY1WZGagpYtTgbWBWGdoUD2CtrPj1mQPjY8Ldc',
    dot: 340282366920938463463374607431768211455n,
    astr: { Concrete: { parents: 0, interior: 'Here' } },
    aca: 18446744073709551616n,
    usdt: 4294969280n,
  },
  shiden: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551616',

    paraAccount: 'F7fq1jNy74AqkJ1DP4KqSrWtnTGtXfNVoDwFhTvvPxUvJaq',
    ksm: 340282366920938463463374607431768211455n,
    sdn: { Concrete: { parents: 0, interior: 'Here' } },
    kar: 18446744073709551618n,
    usdt: 4294969280n,
  },
}

const getInitStorages = (config: typeof custom.astar | typeof custom.shiden) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: '100000000000000000000' } }]],
  },
  Assets: {
    account: [
      [[config.relayToken, defaultAccounts.alice.address], { balance: 10 * 1e12 }],
      [[config.aUSDToken, defaultAccounts.alice.address], { balance: 10 * 1e12 }],
    ],
  },
  Sudo: {
    key: defaultAccounts.alice.address,
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers', 'supportedVersion'],
  },
})

export const astar = defineChain({
  name: 'astar',
  paraId: 2006,
  endpoint: 'wss://astar-rpc.dwellir.com',
  custom: custom.astar,
  initStorages: getInitStorages(custom.astar),
})

export const shiden = defineChain({
  name: 'shiden',
  paraId: 2007,
  endpoint: 'wss://shiden-rpc.dwellir.com',
  custom: custom.shiden,
  initStorages: getInitStorages(custom.shiden),
})
