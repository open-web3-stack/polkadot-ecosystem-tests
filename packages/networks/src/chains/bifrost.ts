import { defineChain } from '../defineChain.js'
import { defaultAccounts, defaultAccountsSr25519, testAccounts } from '../testAccounts.js'

const custom = {
  bifrostPolkadot: {
    relayToken: 'DOT',
    bnc: { Native: 'BNC' },
  },
  bifrostKusama: {
    relayToken: 'KSM',
    bnc: { Token: 'BNC' },
  },
}

const getInitStorages = (_config: typeof custom.bifrostPolkadot | typeof custom.bifrostKusama) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 666e12 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 666e12 } }],
      [[testAccounts.alice.address], { providers: 1, data: { free: 666e12 } }],
    ],
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers'],
  },
})

export const bifrostPolkadot = defineChain({
  name: 'bifrostPolkadot',
  endpoint: 'wss://bifrost-polkadot.ibp.network',
  paraId: 2030,
  custom: custom.bifrostPolkadot,
  initStorages: getInitStorages(custom.bifrostPolkadot),
})

export const bifrostKusama = defineChain({
  name: 'bifrostKusama',
  endpoint: 'wss://us.bifrost-rpc.liebi.com/ws',
  paraId: 2001,
  custom: custom.bifrostKusama,
  initStorages: getInitStorages(custom.bifrostKusama),
})
