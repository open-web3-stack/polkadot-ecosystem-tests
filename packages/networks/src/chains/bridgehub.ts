import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  bridgeHubPolkadot: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
  },
  bridgeHubKusama: {
    ksm: { Concrete: { parents: 1, interior: 'Here' } },
  },
}

const getInitStorages = (_config: typeof custom.bridgeHubPolkadot | typeof custom.bridgeHubKusama) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }]],
  },
})

export const bridgeHubPolkadot = defineChain({
  name: 'bridgeHubPolkadot',
  endpoint: 'wss://polkadot-bridge-hub-rpc.polkadot.io',
  paraId: 1002,
  custom: custom.bridgeHubPolkadot,
  initStorages: getInitStorages(custom.bridgeHubPolkadot),
})

export const bridgeHubKusama = defineChain({
  name: 'bridgeHubKusama',
  endpoint: 'wss://kusama-bridge-hub-rpc.polkadot.io',
  paraId: 1002,
  custom: custom.bridgeHubKusama,
  initStorages: getInitStorages(custom.bridgeHubKusama),
})
