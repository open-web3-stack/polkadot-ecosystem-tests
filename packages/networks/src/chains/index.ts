export * from './acala.js'
export * from './assethub.js'
export * from './astar.js'
export * from './hydraDX.js'
export * from './moonbeam.js'
export * from './polkadot.js'
