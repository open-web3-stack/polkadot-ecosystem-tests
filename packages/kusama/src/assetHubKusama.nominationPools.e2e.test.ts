import { assetHubKusama } from '@e2e-test/networks/chains'
import { baseNominationPoolsE2ETests, registerTestTree, type TestConfig } from '@e2e-test/shared'

const pAssetHubTestConfig: TestConfig = {
  testSuiteName: 'Kusama Asset Hub Nomination Pools',
  addressEncoding: 2,
  blockProvider: 'NonLocal',
  asyncBacking: 'Enabled',
}

registerTestTree(baseNominationPoolsE2ETests(assetHubKusama, pAssetHubTestConfig))
