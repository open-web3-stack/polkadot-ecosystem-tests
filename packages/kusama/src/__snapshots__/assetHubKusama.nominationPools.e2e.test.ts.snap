// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Kusama Asset Hub Nomination Pools > nomination pool creation with insufficient funds > create nomination pool with insufficient funds events 1`] = `
[
  {
    "data": {
      "dispatchError": {
        "Module": {
          "error": "0x08000000",
          "index": 80,
        },
      },
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 20000)",
          "refTime": "(rounded 3000000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pool double join test: an account can only ever be in one pool at a time > join nomination pool events 1`] = `
[
  {
    "data": {
      "amount": "(rounded 1700000000)",
    },
    "method": "Bonded",
    "section": "staking",
  },
  {
    "data": {
      "bonded": "(rounded 1700000000)",
      "joined": true,
      "member": "Cdbxb6S9i8s4swqGZRWhcZyrcaXmTw1MkzLjbUiADjdSo2f",
    },
    "method": "Bonded",
    "section": "nominationPools",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pool double join test: an account can only ever be in one pool at a time > join second nomination pool events 1`] = `
[
  {
    "data": {
      "dispatchError": {
        "Module": {
          "error": "0x04000000",
          "index": 80,
        },
      },
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 20000)",
          "refTime": "(rounded 3000000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pool global config test > setting global nomination pool configs with signed origin 1`] = `
[
  {
    "data": {
      "dispatchError": "BadOrigin",
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 10000)",
          "refTime": "(rounded 1000000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pool global config test > system events 1`] = `
[
  {
    "data": {
      "globalMaxCommission": "10.00%",
      "maxMembers": 2,
      "maxMembersPerPool": 2,
      "maxPools": 2,
      "minCreateBond": "(rounded 1000000000000)",
      "minJoinBond": "(rounded 1700000000)",
    },
    "method": "GlobalParamsUpdated",
    "section": "nominationPools",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pool global config test > system events 2`] = `
[
  {
    "data": {
      "globalMaxCommission": "10.00%",
      "maxMembers": 3,
      "maxMembersPerPool": 3,
      "maxPools": 3,
      "minCreateBond": "(rounded 1000000000000)",
      "minJoinBond": "(rounded 1700000000)",
    },
    "method": "GlobalParamsUpdated",
    "section": "nominationPools",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pool lifecycle test > bond extra funds events 1`] = `
[
  {
    "data": {
      "amount": "(rounded 1000000000000)",
    },
    "method": "Bonded",
    "section": "staking",
  },
  {
    "data": {
      "bonded": "(rounded 1000000000000)",
      "joined": false,
      "member": "Cdbxb6S9i8s4swqGZRWhcZyrcaXmTw1MkzLjbUiADjdSo2f",
    },
    "method": "Bonded",
    "section": "nominationPools",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pool lifecycle test > chill events 1`] = `
[
  {
    "data": {},
    "method": "Chilled",
    "section": "staking",
  },
  {
    "data": {
      "caller": "D2oHoVDFJGy7iDYcY7fbm82CuNycWaSxCGpsQtLqkrE9JKf",
    },
    "method": "PoolNominatorChilled",
    "section": "nominationPools",
  },
  {
    "data": {
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 10000)",
          "refTime": "(rounded 1000000000)",
        },
      },
    },
    "method": "ExtrinsicSuccess",
    "section": "system",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pool lifecycle test > claim commission events 1`] = `
[
  {
    "data": {
      "dispatchError": {
        "Module": {
          "error": "0x1a000000",
          "index": 80,
        },
      },
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 10000)",
          "refTime": "(rounded 1000000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pool lifecycle test > commission alteration events 1`] = `
[
  {
    "data": {
      "current": [
        "0.10%",
        "Cdbxb6S9i8s4swqGZRWhcZyrcaXmTw1MkzLjbUiADjdSo2f",
      ],
    },
    "method": "PoolCommissionUpdated",
    "section": "nominationPools",
  },
  {
    "data": {
      "maxCommission": "1.00%",
    },
    "method": "PoolMaxCommissionUpdated",
    "section": "nominationPools",
  },
  {
    "data": {
      "changeRate": {
        "maxIncrease": "100.00%",
        "minDelay": 10,
      },
    },
    "method": "PoolCommissionChangeRateUpdated",
    "section": "nominationPools",
  },
  {
    "data": {
      "permission": "Permissionless",
    },
    "method": "PoolCommissionClaimPermissionUpdated",
    "section": "nominationPools",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pool lifecycle test > create nomination pool events 1`] = `
[
  {
    "data": {
      "amount": 1000000000000,
    },
    "method": "Bonded",
    "section": "staking",
  },
  {
    "data": {
      "depositor": "D8ew585BL5H1ALhn4kmJoxhgqcgKDZLPc6xunJNv4mmrBns",
    },
    "method": "Created",
    "section": "nominationPools",
  },
  {
    "data": {
      "bonded": 1000000000000,
      "joined": true,
      "member": "D8ew585BL5H1ALhn4kmJoxhgqcgKDZLPc6xunJNv4mmrBns",
    },
    "method": "Bonded",
    "section": "nominationPools",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pool lifecycle test > join nomination pool events 1`] = `
[
  {
    "data": {
      "amount": 1000000000000,
    },
    "method": "Bonded",
    "section": "staking",
  },
  {
    "data": {
      "bonded": 1000000000000,
      "joined": true,
      "member": "Cdbxb6S9i8s4swqGZRWhcZyrcaXmTw1MkzLjbUiADjdSo2f",
    },
    "method": "Bonded",
    "section": "nominationPools",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pool lifecycle test > nomination pool validator selection events 1`] = `
[
  {
    "data": {
      "caller": "D2oHoVDFJGy7iDYcY7fbm82CuNycWaSxCGpsQtLqkrE9JKf",
    },
    "method": "PoolNominationMade",
    "section": "nominationPools",
  },
  {
    "data": {
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 50000)",
          "refTime": "(rounded 2000000000)",
        },
      },
    },
    "method": "ExtrinsicSuccess",
    "section": "system",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pool lifecycle test > set state events 1`] = `
[
  {
    "data": {
      "newState": "Blocked",
    },
    "method": "StateChanged",
    "section": "nominationPools",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pool lifecycle test > set state to destroying events 1`] = `
[
  {
    "data": {
      "newState": "Destroying",
    },
    "method": "StateChanged",
    "section": "nominationPools",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pool lifecycle test > unbond (depositor) events 1`] = `
[
  {
    "data": {
      "dispatchError": {
        "Module": {
          "error": "0x08000000",
          "index": 80,
        },
      },
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 40000)",
          "refTime": "(rounded 3000000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pool lifecycle test > unbond (kick) events 1`] = `
[
  {
    "data": {
      "amount": 1000000000000,
    },
    "method": "Unbonded",
    "section": "staking",
  },
  {
    "data": {
      "balance": 1000000000000,
      "member": "Cdbxb6S9i8s4swqGZRWhcZyrcaXmTw1MkzLjbUiADjdSo2f",
      "points": 1000000000000,
    },
    "method": "Unbonded",
    "section": "nominationPools",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pool lifecycle test > unbond events 1`] = `
[
  {
    "data": {
      "amount": "(rounded 1000000000000)",
    },
    "method": "Unbonded",
    "section": "staking",
  },
  {
    "data": {
      "balance": "(rounded 1000000000000)",
      "member": "Cdbxb6S9i8s4swqGZRWhcZyrcaXmTw1MkzLjbUiADjdSo2f",
      "points": "(rounded 1000000000000)",
    },
    "method": "Unbonded",
    "section": "nominationPools",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pool lifecycle test > update roles events 1`] = `
[
  {
    "data": {
      "bouncer": "H7L1Xh1XZxbKktdoEG5rkWdUdydq9e3NpsjmcSHpUHM8Z6h",
      "nominator": "D2oHoVDFJGy7iDYcY7fbm82CuNycWaSxCGpsQtLqkrE9JKf",
      "root": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
    },
    "method": "RolesUpdated",
    "section": "nominationPools",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pool metadata test > create nomination pool events 1`] = `
[
  {
    "data": {
      "amount": 1000000000000,
    },
    "method": "Bonded",
    "section": "staking",
  },
  {
    "data": {
      "depositor": "D8ew585BL5H1ALhn4kmJoxhgqcgKDZLPc6xunJNv4mmrBns",
    },
    "method": "Created",
    "section": "nominationPools",
  },
  {
    "data": {
      "bonded": 1000000000000,
      "joined": true,
      "member": "D8ew585BL5H1ALhn4kmJoxhgqcgKDZLPc6xunJNv4mmrBns",
    },
    "method": "Bonded",
    "section": "nominationPools",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pool metadata test > set metadata events 1`] = `
[
  {
    "data": {
      "caller": "D8ew585BL5H1ALhn4kmJoxhgqcgKDZLPc6xunJNv4mmrBns",
    },
    "method": "MetadataUpdated",
    "section": "nominationPools",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pools update roles test > update pool roles via scheduler pallet 1`] = `
[
  {
    "event": {
      "data": [
        "D2oHoVDFJGy7iDYcY7fbm82CuNycWaSxCGpsQtLqkrE9JKf",
        "Cdbxb6S9i8s4swqGZRWhcZyrcaXmTw1MkzLjbUiADjdSo2f",
        "H7L1Xh1XZxbKktdoEG5rkWdUdydq9e3NpsjmcSHpUHM8Z6h",
      ],
      "index": "0x5008",
    },
    "phase": {
      "initialization": null,
    },
    "topics": [],
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pools update roles test > update roles events 1`] = `
[
  {
    "data": {
      "bouncer": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
      "nominator": "H7L1Xh1XZxbKktdoEG5rkWdUdydq9e3NpsjmcSHpUHM8Z6h",
      "root": "D8ew585BL5H1ALhn4kmJoxhgqcgKDZLPc6xunJNv4mmrBns",
    },
    "method": "RolesUpdated",
    "section": "nominationPools",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pools update roles test > update roles failure events 1`] = `
[
  {
    "data": {
      "dispatchError": {
        "Module": {
          "error": "0x11000000",
          "index": 80,
        },
      },
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 10000)",
          "refTime": "(rounded 800000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Kusama Asset Hub Nomination Pools > nomination pools update roles test > update roles remove self events 1`] = `
[
  {
    "data": {
      "bouncer": "EvvS4HELEC1nxVHsjYUsPvypd2kXeu78AguyiSN7kV2pTnk",
      "nominator": "H7L1Xh1XZxbKktdoEG5rkWdUdydq9e3NpsjmcSHpUHM8Z6h",
      "root": null,
    },
    "method": "RolesUpdated",
    "section": "nominationPools",
  },
]
`;
